packages/lb-components/src/components/videoPlayer/TagToolInstanceAdaptor.tsx

```diff
@@ -0,0 +1,308 @@
+  IVideoTagInstanceAdaptorState
+> {
+  public fns: { [key: string]: () => void } = {};
+  public videoRef?: HTMLVideoElement;
+  public labelSelectedList: number[] = [];
+
+  constructor(props: IVideoTagInstanceAdaptorProps) {
+    super(props);
+    this.state = {
+      tagResult: [],
+      labelSelectedList: [],
+      valid: true,
+    };
+  }
+
+  public get config() {
+    const stepInfo = StepUtils.getCurrentStepInfo(this.props.step, this.props.stepList);
+    return jsonParser(stepInfo?.config);
+  }
+
+  /** Just implementation, no actual logic */
+  public get history() {
+    return { initRecord: () => {}, pushHistory: () => {} };
+  }
+
+  get currentTagResult() {
+    return this.state.tagResult[0];
+  }
+
+  public get valid() {
+    return this.state.valid;
+  }
+
+  public clearResult = (sendMsg: boolean = true, value?: string) => {
+    const newTag = value
+      ? this.state.tagResult.map((v) => {
+          if (v?.result[value]) {
+            delete v.result[value];
+          }
+          return v;
+        })
+      : [];
+
+    this.setState(
+      {
+        tagResult: newTag,
+      },
+      () => this.emitEvent('render'),
+    );
+  };
+
+  public exportData = () => {
+    const duration = this.videoRef?.duration ?? 0;
+    const videoQulity = this.videoRef?.getVideoPlaybackQuality();
+    const frames = videoQulity?.totalVideoFrames;
+    const videoWidth = this.videoRef?.videoWidth ?? 0;
+    const videoHeight = this.videoRef?.videoHeight ?? 0;
+
+    return [
+      this.state.tagResult,
+      { valid: this.state.valid, duration, frames, videoWidth, videoHeight },
+    ];
+  };
+
+  public singleOn(event: string, func: () => void) {
+    this.fns[event] = func;
+  }
+
+  public on(event: string, func: () => void) {
+    this.singleOn(event, func);
+  }
+
+  public getTagResultByCode(num1: number, num2?: number) {
+    try {
+      const inputList = this.config?.inputList ?? [];
+      const mulitTags = inputList.length > 1;
+      const keycode1 = num2 !== undefined ? num1 : 0;
+      const keycode2 = num2 !== undefined ? num2 : num1;
+      const primaryTagConfig = mulitTags ? inputList[keycode1] : inputList[0];
+      const secondaryTagConfig = (primaryTagConfig.subSelected ?? [])[keycode2];
+
+      if (primaryTagConfig && secondaryTagConfig) {
+        return {
+          value: {
+            key: primaryTagConfig.value,
+            value: secondaryTagConfig.value,
+          },
+          isMulti: primaryTagConfig.isMulti,
+        };
+      }
+    } catch {
+      return;
+    }
+  }
+
+  public setLabelBySelectedList(num1: number, num2?: number) {
+    const newTagConfig = this.getTagResultByCode(num1, num2);
+
+    if (newTagConfig) {
+      const tagRes = this.combineResult(newTagConfig, this.state.tagResult[0]?.result ?? {});
+
+      const tagResult = [
+        {
+          sourceID: CommonToolUtils.getSourceID(),
+          id: uuid(8, 62),
+          result: tagRes,
+        },
+      ];
```

Comment:
This way will make different result with `id` every times. You need to fix it. It needs to be init when the result is emoty